cirq/contrib/quantum_volume/quantum_volume.py

```diff
@@ -162,34 +237,55 @@ def compile_circuit(
 
     """
     compiled_circuit = circuit.copy()
+
+    # Optionally add some the parity check bits.
+    parity_map: Dict[cirq.Qid, cirq.Qid] = {}  # original -> parity
+    if add_readout_error_correction:
+        parity_ops = cirq.Moment()
+        # Sort just to make it deterministic.
+        for idx, qubit in enumerate(sorted(compiled_circuit.all_qubits())):
+            # For each qubit, create a new qubit that will serve as its parity
+            # check. This parity bit is initialized to 0 and then CNOTed with
```

Comment:
readout error is asymmetric: p(0|0) > p(1|1). You can account for that by encoding into the single-excitation subspace; i.e. use an inverse-controlled not: `X CNOT X` to encode 0 -> 01 and 1 -> 10 so you (should) always readout one 1

Reply:
Neat, I did not know that - Fixed. Out of curiosity, why is readout error asymmetric? Is that just for our device, or for all devices?

Also, probably a dumb question, but could I instead do `X(ancilla) CNOT(physical, ancilla)` instead of `X(physical) CNOT(physical, ancilla) X(physical)` ?

Reply:
Readout takes longer than gates (~1000ns vs 10) so the qubit can undergo T1 decay during readout (see cirq.contrib.noise_models). I think it's fairly common in superconducting architectures like ours. 

You could do the suggested alternate, but please make sure that the X(ancilla) is done as late as possible. If cirq is left to its own devices, the X will happen at the beginning, the qubit will be hanging out in |1> and could more readily undergo T1 decay